StripePaymentSheet/StripePaymentSheet/Source/Helpers/SavedPaymentMethodManager.swift

```diff
@@ -6,26 +6,52 @@
 //
 
 import Foundation
+@_spi(STP) import StripeCore
 @_spi(STP) import StripePayments
 
 /// Provides shared implementations of common operations for managing saved payment methods in PaymentSheet
 final class SavedPaymentMethodManager {
 
+    enum Error: Swift.Error {
+        case missingEphemeralKey
+    }
+
     let configuration: PaymentSheet.Configuration
+    let intent: Intent
+
+    private var ephemeralKey: String? {
+        guard let ephemeralKey = configuration.customer?.ephemeralKeySecretBasedOn(intent: intent) else {
+            stpAssert(true, "Failed to read ephemeral key.")
+            let errorAnalytic = ErrorAnalytic(event: .unexpectedPaymentSheetError,
+                                              error: Error.missingEphemeralKey,
+                                              additionalNonPIIParams: ["customer_access_provider": configuration.customer?.customerAccessProvider.analyticValue as Any])
+            STPAnalyticsClient.sharedClient.log(analytic: errorAnalytic)
+            return nil
+        }
+        return ephemeralKey
```

Comment:
nit: Why compute this?  Seems like it should be initialized once and send the error once?